library/std/src/io/buffered/bufreader.rs

```diff
@@ -259,6 +259,31 @@ impl<R: ?Sized + Seek> BufReader<R> {
     }
 }
 
+impl<R> BufReader<R>
+where
+    Self: Read,
```

Comment:
Any reason for this impl block instead of putting the `impl SpecReadByte for BufReader` in this module?

Reply:
Good suggestion!